options/auto_retry_options.go

```diff
@@ -18,3 +18,11 @@ var RETRYABLE_ERRORS = []string{
 	"NoSuchBucket: The specified bucket does not exist",
 	"(?s).*Error creating SSM parameter: TooManyUpdates:.*",
 }
+
+// List of recurring transient errors encountered when calling terraform
+// If one of these are encountered, we will re-run the init command
+var ERRORS_REQUIRING_INIT = []string{
+	"(?s).*Module version requirements have changed.*",
+	"(?s).*Could not satisfy plugin requirements.*",
+	"(?s).*Error accessing remote module registry.*",
```

Comment:
What about:

```
 * provider.external: new or changed plugin executable
```

I used to get this all the time. Not sure if that error message changed?

And:

```
Please run "terraform init".
```

This is part of the "Could not satisfy plugin requirements" message you have above, but perhaps adding it too is a good way to catch other similar errors, where Terraform clearly knows what needs to be done, but for some reason chooses not to do it?

Reply:
There are more error messages to be added to both the `RETRYABLE_ERRORS ` and `ERRORS_REQUIRING_INIT `. I will take care about it. The one I added here are mainly those I was able to use when resting (reproducing) the problem locally. Update is coming shortly.

Reply:
I added a number of new error messages to match against - hopefully these would cover most of the common errors.